Introduce unique prefix for Redmew GUI elements

Soggs opened the pull request:
Due to collisions with mods when naming GUI elements with just numbers, I added a prefix when naming elements with ``utils.token.uid()``.

Soggs committed:
Introduce unique prefix for Redmew GUI elements

Due to collisions with mods when naming GUI elements with just numbers, I added a prefix when naming elements with ``utils.token.uid()``.

grilledham commented on utils/gui.lua:
```diff
@@ -6,6 +6,8 @@ local Styles = require 'resources.styles'
 local tostring = tostring
 local next = next
 
+local gui_element_prefix = "Redmew_GUI_element_"
```
Maybe we could go shorter like just `"Redmew_"`

Soggs replied:
We can definitely shorten it. I wanted to be more verbose rather than less to start out.

grilledham reviewed (commented):
Have you tested this fixes the problem you were having when running mods?

Soggs commented:
Yeah I tested it. It fixes the problems with Module Inserter. I am still uncertain however why it fired in the first place and also why it only fires both with the MI buttons but not with the Redmew GUI button.

I am at the same time investigating how I could change the MI buttons but the GUI there uses the ``name`` strings as numeric indices in a few places, which I hope we do not do anywhere. Though I think you should never do something like that in the first place and other elements in our GUI do use strings and not numbers like the diggy exp button is internally called ``Diggy.Experience.Button``.

Soggs committed:
Update gui.lua

Shortened the prefix to ``Redmew_`` as per comment: https://github.com/Refactorio/RedMew/pull/1297#discussion_r825332134

grilledham committed:
Fix gui_element_prefix not working in debug mode.
